### PR TITLE
Mayh 10179 asif support for bitmap

### PR DESCRIPTION
**Pull request by jianwan2016**
Add `BitMap` type for asif.

### CODE REVIEW

File: app/App/Commands/Dump.hs [hunk 1]
```diff
@@ -162,6 +162,8 @@ runDump opt = do
             liftIO $ LBSC.hPutStrLn hOut (segment ^. the @"payload")
           Just (Known F.BitString) ->
             liftIO $ IO.hPutStrLn hOut (bitShow (segment ^. the @"payload"))
+          Just (Known F.BitMap) ->
+            liftIO $ IO.hPutStrLn hOut (bitShow (segment ^. the @"payload"))
```
Comment (dsturnbull) [inline app/App/Commands/Dump.hs]: Is this going to print 4GB of 1s and 0s?

**Comment by newhoggy** [comment]
Can you paste example output?

**Comment by jianwan2016** [comment]
This is the output:
```
EC2
EC2
EC2
EC2
==== [6] akamai ====
1
```

**Comment by jianwan2016** [comment]
```
==== [6] akamai ====
0.0.0.63        (63)
```

File: app/App/Commands/Dump.hs [hunk 2]
```diff
@@ -172,4 +178,14 @@ runDump opt = do
               liftIO $ IO.hPutStr hOut $ (\c -> if isPrint c then c else '.') <$> LBSC.unpack bs
               liftIO $ IO.hPutStrLn hOut ""
 
-  where magic = AP.string "seg:" *> (BS.pack <$> many AP.anyWord8) AP.<?> "\"seg:????\""
+  where
+    magic = AP.string "seg:" *> (BS.pack <$> many AP.anyWord8) AP.<?> "\"seg:????\""
+    bitsToW32s :: Int -> (Int, Word64) -> [Word32] -> [Word32]
+    bitsToW32s bp (i, w64) acc =
```
Comment (dsturnbull) [inline app/App/Commands/Dump.hs]: Maybe `word64ToList` would be useful to reuse?

Comment (jianwan2016) [reply app/App/Commands/Dump.hs]:
```λ> word64ToList 32 0x003300 []
[0.0.8.8,0.0.8.9,0.0.8.12,0.0.8.13]
it :: [HaskellWorks.Data.Network.Ip.Type.Ipv4Address]
(0.18 secs, 1,079,528 bytes)
λ> bitsToW32s 0 (32, 0x003300) []
[0.0.8.50,0.0.8.51,0.0.8.54,0.0.8.55]
it :: [HaskellWorks.Data.Network.Ip.Type.Ipv4Address]
(0.02 secs, 1,105,296 bytes)
λ>
```

**Pull request merged**